Kernel/Devices/GPU/Console/GenericFramebufferConsole.cpp

```diff
@@ -322,12 +342,19 @@ void GenericFramebufferConsoleImpl::write(size_t x, size_t y, char ch, Color bac
         offset_in_framebuffer.bytes += framebuffer_pitch();
     }
     flush_glyph(x, y);
+
     m_x = x + 1;
     if (m_x >= max_column()) {
         m_x = 0;
         m_y = y + 1;
-        if (m_y >= max_row())
-            m_y = 0;
+        if (m_y >= max_row()) {
+            if (critical) {
+                m_y = max_row() - 1;
+                scroll_up();
+            } else {
```

Comment:
What happens in this case? When we aren't in critical mode, shouldn't we scroll as well?

Reply:
Seems good then, thanks for the clarification!